src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Umbraco.Core.Events;
+using Umbraco.Core.Logging;
+using Umbraco.Web.Common.Lifetime;
+
+namespace Umbraco.Web.Common.Profiler
+{
+    public class InitializeWebProfiling : INotificationHandler<UmbracoApplicationStarting>
+    {
+        private readonly bool _profile;
+        private readonly WebProfiler _profiler;
+        private readonly IUmbracoRequestLifetime _umbracoRequestLifetime;
+        private readonly List<Action> _terminate = new List<Action>();
```

Comment:
Adding stuff to _terminate doesn't seem to do anything now

Reply:
Damn, of course not..  As har as I know, we don't really have a good way to handle this? 
1. We can use the `UmbracoApplicationStopping` event, but then we don't really has the connection between the two notification handlers?
2. We could change the IUmbracoRequestLifetime.RequestStart event to be handled by the EventAggregator.